tracks/python/exercises/isbn-verifier/mentoring.md

```diff
@@ -0,0 +1,111 @@
+# Mentoring
+
+## Problem and Challenges
+
+The problem asks you to write an ISBN validation checker.
+
+* Any "-" in the input string should be ignored.
+* A valid ISBN must be 10 values long.
+* A valid ISBN must be entirely digits, other than the last character.
+* A valid ISBN may have `"X"` as the last character, which should be treated as a `10` value.
+* The ISBN is valid if `sum(value * position_counter) % 11 == 0` for `value in <input string>` and `position_counter in <(10..1)>`.
+
+## Reasonable Solution
+
+```python
+def is_valid(isbn: str) -> bool:
+    values = [value for value in isbn if value != "-"]
+
+    if len(values) != 10:
+        return False
+
+    if values[-1] == "X":
+        values[-1] = "10"
+
+    if any(not value.isdigit() for value in values):
+        return False
+
+    total = sum(int(value) * (10 - position) for position, value in enumerate(values))
+    return total % 11 == 0
+```
+
+## Talking Points
+
+### Generator expressions
+
+The sum can be computed using a `for` loop or via the `sum()` [built-in function][builtins-functions].
+
+```python
+total = 0
+for position, value in enumerate(values)):
+    total += int(value) * (10 - position)
+```
+
+Generator expressions are considered very Pythonic and enable compact and clean iteration.
+They are also very efficient.
+
+### The position value
+
+The position value can be directly generated using `range()`.
+However, using `range()` requires mapping the position value to an ISBN positional value, by computing an index or via `zip()`.
+
+```python
+total = 0
+for position, value in zip(range(10, 0, -1), values):
+    total += int(value) * position
+```
+
+The [`enumerate()` function][builtins-functions] provides a nice way to get both a position and a value.
+The position returned by `enumerate()` does need to be modified (`10 - position`) to be used.
+
+### The trailing "X" value
+
+The final value can be handled in the summation loop by checking the position and the value and replacing "X" with 10.
+Doing so allows for keeping the ISBN as a `str` instead of converting it to `list[str]` to be able to hold a `"10"` value.
+
+```python
+total = 0
+for position, value in enumerate(isbn):
+    if position == 9 and value == "X":
+        total += 10
+    else:
+        total += int(value) * (10 - position)
+```
+
+Recognizing that changing the data from `str` to `list[str]` allows for more flexibility of what it can store (`"10"`) and a simpler summation loop.
+
+### Checking for non-digits
+
+The [built-in `any()` function][builtin-functions] is very convenient for any non-digit values in the input.
```

Comment:
It is very convenient for any non-digit values in the input, but why?  How is it convenient?  What does it do for it?  It feels like there might be a word missing or perhaps a couple of words in this sentence.

Reply:
What would you like to see here? How about:
```suggestion
The [built-in `any()` function][builtin-functions] is a concise approach for checking for any non-digit values in the input.
```

Reply:
That makes more sense to me.